### ISSUE TITLE
[Release]: v0.4.1

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/teal.widgets#189]
### Issues

- [X] Update README as directed [here](https://github.com/insightsengineering/coredev-tasks/issues/462)
    * https://github.com/insightsengineering/teal.widgets/pull/191
    * https://github.com/insightsengineering/teal.widgets/pull/193/
- [X] Group vignettes in a meaningful cohorts as directed [here](https://github.com/insightsengineering/coredev-tasks/issues/468)
    * No need, as there are only 2 vignettes
- [x] Remove `:::` from examples https://github.com/insightsengineering/teal.widgets/pull/194
- [x] https://github.com/insightsengineering/teal.widgets/issues/136 https://github.com/insightsengineering/teal.widgets/pull/163

### Pre-requisites

- [x] Make sure that high priority bugs (label "priority" + "bug") have been resolved before going into the release.
- [x] Review old/hanging PRs before going into the release (Optional).
- [x] Revisit R-package's lifecycle badges (Optional).
- [X] Make sure that all upstream dependencies of this package that need to be submitted to CRAN were accepted before going into release activities.
- [x] Make sure integration tests are green 2-3 days before the release. Look carefully through logs (check for warnings and notes).
- [x] Decide what gets merged in before starting release activities.

### Release Checklist

- [x] Update NEWS.md file: make sure it reflects a holistic summary of what has changed in the package.
- [X] Remove the additional fields (`Remotes` ~`and `Config/Needs/*~) from the DESCRIPTION file where applicable.
- [X] Increase versioned dependency on {package name} to >=X.X.X (Optional).
- [x] Make sure that the minimum dependency versions are updated in the DESCRIPTION file for the package and its reverse dependencies (Optional).
- [x] Create a pull request to make necessary bug fixes/changes (add "[skip vbump]" in the pr title), and after merging the PR, tag the update(s) as a release candidate v < intended release version > -rc < release candidate iteration > on the main branch.
- [x] Build the package locally using the command:`R CMD build .` which will generate a .tar.gz file necessary for the CRAN submission.
- [x] Submit the package that was build in the previous step via this form: https://cran.r-project.org/submit.html.
- [x] Address CRAN feedback, tag the package vX.X.X-rc(n+1) and repeat the submission to CRAN whenever necessary.
- [x] Get the package accepted and published on CRAN.
- [x] If the additional fields were removed, add them back in a separate PR, and then merge the PR back to main. Note: Take precautionary measures to ensure that the version bump does not take place on a merge.
- [x] Create a git tag with the final version set to X.X.X on the main branch.

### Testing

- [x] Integration tests results - accepted: https://github.com/insightsengineering/idr-tasks/issues/673
- [x] UAT results - accepted. (Sanity check by playing around with the example app locally)
- [x] All testing activities are finalized.

### Release Feedback (Not Applicable)

- [ ] ~~Fix 1~~
- [ ] ~~Enhancement 1~~
- [ ] ~~Defect 1~~

### Post-release Checklist

- [ ] Make sure that the package is published to internal repositories.
- [ ] Review and update installation instructions for the package wherever needed (Optional).
- [ ] Update all integration tests to reference the new release.
- [ ] Ensure a new dev version (.9XXX) is added to the NEWS.md file and DESCRIPTION file as a placeholder for release notes.
- [ ] Announce the release on ________.

### Decision tree

Click [here](https://github.com/insightsengineering/.github/blob/main/.github/ISSUE_TEMPLATE/RELEASE_DECISION_TREE.md) to see the release decision tree.

**Comment by donyunardi** [comment]
There are several examples using `:::`. Will need to fix this before submitting to CRAN.

**Comment by m7pr** [comment]
Hey @donyunardi I provided few small PRs with a little cleaneup and assigned you as a reviewer. Let me know if you'd need any more help releasing while the package?

**Comment by donyunardi** [comment]
> Hey @donyunardi I provided few small PRs with a little cleaneup and assigned you as a reviewer. Let me know if you'd need any more help releasing while the package?

Thanks @m7pr!

**Comment by donyunardi** [comment]
package submitted to CRAN:
![image](https://github.com/insightsengineering/teal.widgets/assets/8597300/a50bfff4-02bd-4e27-a96b-18371d9d23ff)

heads up @gogonzo

**Comment by gogonzo** [comment]
```
Thanks,

If there are references describing the methods in your package, please
add these in the description field of your DESCRIPTION file in the form
authors (year) <doi:...>
authors (year) <arXiv:...>
authors (year, ISBN:...)
or if those are not available: <https:...>
with no space after 'doi:', 'arXiv:', 'https:' and angle brackets for
auto-linking.
(If you want to add a title as well please put it in quotes: "Title")

Please add \value to .Rd files regarding exported methods and explain
the functions results in the documentation. Please write about the
structure of the output (class) and also what the output means. (If a
function does not return a value, please document that too, e.g.
\value{No return value, called for side effects} or similar)
Missing Rd-tags:
      get_dt_rows.Rd: \value
      optionalSelectInput.Rd: \value
      optionalSliderInput.Rd: \value
      optionalSliderInputValMinMax.Rd: \value
      plot_with_settings_srv.Rd: \value
      plot_with_settings_ui.Rd: \value
      table_with_settings.Rd: \value

Functions which are supposed to only run interactively (e.g. shiny)
should be wrapped in if(interactive()).
Please replace /dontrun{} with if(interactive()){} if possible,
then users can see that the functions are not intended for use in
scripts / functions that are supposed to run non interactively.
```

**Comment by gogonzo** [comment]
I've made a commit with the response to the comments. I've:
- added @return to the functions
- merged documentation of srv_plot_with_settings and ui_plot_with_settings
- replaced all dontrun -> if (interative()) 
- fixed not executable example of get_dt_rows

**Comment by donyunardi** [comment]
Thanks @gogonzo 
Made some small changes and resubmitted to CRAN.
Take 2.
